test/gracefulstop_test.go

```diff
@@ -113,31 +113,27 @@ func (s) TestGracefulStop(t *testing.T) {
 	d := func(ctx context.Context, _ string) (net.Conn, error) { return dlis.Dial(ctx) }
 
 	ss := &stubserver.StubServer{
+		Listener: dlis,
 		FullDuplexCallF: func(stream testgrpc.TestService_FullDuplexCallServer) error {
 			_, err := stream.Recv()
 			if err != nil {
 				return err
 			}
 			return stream.Send(&testpb.StreamingOutputCallResponse{})
 		},
+		S: grpc.NewServer(),
 	}
-	s := grpc.NewServer()
-	testgrpc.RegisterTestServiceServer(s, ss)
+	stubserver.StartTestService(t, ss)
```

Comment:
I don't think this is doing the correct thing. What we want as part of this test is for `GracefulStop` to be called when `Accept` is in the middle of accepting a connection.

If you folks have convinced yourself that it actually does what it is supposed to do, then we don't need a `sync.WaitGroup`. The same can be accomplished with a `done` channel.

Reply:
Removed sync.WaitGroup